consensus/spos/bls/v2/subroundEndRound_test.go

```diff
@@ -776,7 +788,18 @@ func TestSubroundEndRound_ReceivedProof(t *testing.T) {
 
 		sr.ReceivedProof(proof)
 	})
-	t.Run("should return false when final info is not valid", func(t *testing.T) {
+	t.Run("should early return when header is not for current consensus", func(t *testing.T) {
+		t.Parallel()
+
+		hdr := &block.Header{Nonce: 37}
+		sr := initSubroundEndRound(&statusHandler.AppStatusHandlerStub{})
+		sr.SetHeader(hdr)
+		sr.AddReceivedHeader(hdr)
+
+		proof := &block.HeaderProof{}
+		sr.ReceivedProof(proof)
+	})
```

Comment:
can we add an additional check in these tests? that `CommitBlock` was not called or something

Reply:
pushed